src/ast/mod.rs

```diff
@@ -669,12 +669,21 @@ pub enum Expr {
     /// ```sql
     /// STRUCT<[field_name] field_type, ...>( expr1 [, ... ])
     /// ```
+    /// `DuckDB` specific `Struct` literal expression [2]
+    /// Syntax:
+    /// ```sql
+    /// syntax: {'field_name': expr1[, ... ]}
+    /// ```
     /// [1]: https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#struct_type
+    /// [2]: https://duckdb.org/docs/sql/data_types/struct#creating-structs
     Struct {
         /// Struct values.
         values: Vec<Expr>,
         /// Struct field definitions.
         fields: Vec<StructField>,
+        /// true if uses duckdb array notation syntax(no change in semantics)
+        /// this field is used for formatting
+        array_notation: bool,
```

Comment:
I think it is almost different than the existing struct,  I would prefer we introduce a new Expr.
Note that `Map` in Duckdb has the similar syntax, so I don't think we should name it struct-specific.

I would call it `Expr::Dict` por `Expr::Dictionary`
![Screenshot 2024-03-27 at 08 02 07](https://github.com/sqlparser-rs/sqlparser-rs/assets/15869383/ef146e53-7f73-4412-b31e-31ad4076a37a)